--- a/src/winio.c	2026-04-30 06:12:24.404131816 -0400
+++ b/src/winio.c	2026-04-30 06:14:25.085854416 -0400
@@ -1210,10 +1210,16 @@
 #ifdef __linux__
 	/* When not running under X, check for the bare arrow keys whether
 	 * Shift/Ctrl/Alt are being held together with them. */
-	unsigned char modifiers = 6;
+	unsigned char modifiers = 0;
 
 	/* Modifiers are: Alt (8), Ctrl (4), Shift (1). */
-	if (on_a_vt && !mute_modifiers && ioctl(0, TIOCLINUX, &modifiers) >= 0) {
+	if(GetAsyncKeyState(VK_SHIFT) < 0)
+		modifiers |= 0x01;
+	if(GetAsyncKeyState(VK_CONTROL) < 0)
+		modifiers |= 0x04;
+	if(GetAsyncKeyState(VK_LMENU) < 0)
+		modifiers |= 0x08;
+	if (!mute_modifiers) {
 #ifndef NANO_TINY
 		/* Is Shift being held? */
 		if (modifiers & 0x01) {
