--- a/src/nano.c	2026-04-30 06:17:07.610694176 -0400
+++ b/src/nano.c	2026-04-30 06:22:39.459753625 -0400
@@ -41,6 +41,9 @@
 #ifdef __linux__
 #include <sys/vt.h>
 #endif
+#define WIN32_LEAN_AND_MEAN 1
+#include <windows.h>
+#include <shlobj.h>
 
 #ifdef ENABLE_MULTIBUFFER
 #define read_them_all  TRUE
@@ -230,10 +233,6 @@
 {
 	curs_set(1);
 	endwin();
-#ifndef NANO_TINY
-	printf("\x1B[?2004l");
-	fflush(stdout);
-#endif
 	tcsetattr(STDIN_FILENO, TCSANOW, &original_state);
 }
 
@@ -849,7 +848,7 @@
 /* Reconnect standard input to the tty, and store its state. */
 void reconnect_and_store_state(void)
 {
-	int thetty = open("/dev/tty", O_RDONLY);
+	int thetty = open("CONIN$", O_RDONLY);
 
 	if (thetty < 0 || dup2(thetty, STDIN_FILENO) < 0)
 		die(_("Could not reconnect stdin to keyboard\n"));
@@ -869,14 +868,17 @@
 	restore_terminal();
 
 	/* When input comes from a terminal, show a helpful message. */
-	if (isatty(STDIN_FILENO))
+	if (GetConsoleWindow() != NULL)
 		fprintf(stderr, _("Reading data from keyboard; "
 							"type ^D or ^D^D to finish.\n"));
 
 	/* Open standard input. */
-	stream = fopen("/dev/stdin", "rb");
+	fd = dup(0);
+	stream = fdopen(fd, "rb");
 	if (stream == NULL) {
 		int errnumber = errno;
+		if(fd != -1)
+			close(fd);
 
 		terminal_init();
 		doupdate();
@@ -1242,11 +1244,6 @@
 
 	disable_kb_interrupt();
 
-#ifndef NANO_TINY
-	/* Tell the terminal to enable bracketed pastes. */
-	printf("\x1B[?2004h");
-	fflush(stdout);
-#endif
 }
 
 /* Ask ncurses for a keycode, or assign a default one. */
@@ -2164,13 +2161,8 @@
 		}
 	}
 
-#ifndef NANO_TINY
-	set_up_sigwinch_handler();
-#endif
-
-	/* Curses needs TERM; if it is unset, try falling back to a VT220. */
-	if (getenv("TERM") == NULL)
-		putenv("TERM=vt220");
+	/* Always use the Windows console driver. */
+	putenv("TERM=#win32con");
 
 	/* Enter into curses mode.  Abort if this fails. */
 	if (initscr() == NULL)
@@ -2630,7 +2622,7 @@
 	UNSET(NOREAD_MODE);
 
 	/* Nano is a hands-on editor -- it needs a keyboard. */
-	if (!isatty(STDIN_FILENO))
+	if (GetConsoleWindow() == NULL)
 		die(_("Standard input is not a terminal\n"));
 
 	/* If no filenames were given, or all of them were invalid things like
