diff -ruN nano/src/nano.c nano-p/src/nano.c
--- a/src/nano.c	2026-08-17 13:24:28.645446649 -0400
+++ b/src/nano.c	2026-08-17 13:32:15.845448376 -0400
@@ -230,10 +230,6 @@
 {
 	curs_set(1);
 	endwin();
-#ifndef NANO_TINY
-	printf("\x1B[?2004l");
-	fflush(stdout);
-#endif
 	tcsetattr(STDIN_FILENO, TCSANOW, &original_state);
 }
 
@@ -1230,11 +1226,6 @@
 
 	disable_kb_interrupt();
 
-#ifndef NANO_TINY
-	/* Tell the terminal to enable bracketed pastes. */
-	printf("\x1B[?2004h");
-	fflush(stdout);
-#endif
 }
 
 /* Ask ncurses for a keycode, or assign a default one. */
@@ -2488,12 +2479,6 @@
 	shiftaltright = get_keycode("kRIT4", SHIFT_ALT_RIGHT);
 	shiftaltup = get_keycode("kUP4", SHIFT_ALT_UP);
 	shiftaltdown = get_keycode("kDN4", SHIFT_ALT_DOWN);
-
-#ifdef HAVE_DEFINE_KEY
-	/* Tell ncurses to transform bracketed-paste sequences into keycodes. */
-	define_key("\e[200~", START_OF_PASTE);
-	define_key("\e[201~", END_OF_PASTE);
-#endif
 #endif
 	mousefocusin = get_keycode("kxIN", FOCUS_IN);
 	mousefocusout = get_keycode("kxOUT", FOCUS_OUT);
diff -ruN nano/src/winio.c nano-p/src/winio.c
--- a/src/winio.c	2026-08-17 13:30:32.905447995 -0400
+++ b/src/winio.c	2026-08-17 13:32:57.925448531 -0400
@@ -1531,12 +1531,6 @@
 	if (!ISSET(RAW_SEQUENCES))
 		keypad(frame, FALSE);
 
-#ifndef NANO_TINY
-	/* Turn bracketed-paste mode off. */
-	printf("\x1B[?2004l");
-	fflush(stdout);
-#endif
-
 	linger_after_escape = TRUE;
 
 	/* Read in a single byte or two escapes. */
@@ -1554,12 +1548,6 @@
 
 	linger_after_escape = FALSE;
 
-#ifndef NANO_TINY
-	/* Turn bracketed-paste mode back on. */
-	printf("\x1B[?2004h");
-	fflush(stdout);
-#endif
-
 	/* Turn flow control characters back on if necessary and turn the
 	 * keypad back on if necessary now that we're done. */
 	if (ISSET(PRESERVE))
