From cac0ca725aaecc2779dd0a8d9822002cdcea5601 Mon Sep 17 00:00:00 2001
From: Chawye Hsu <su+git@chawyehsu.com>
Date: Thu, 16 Apr 2026 00:41:05 +0800
Subject: [PATCH 06/17] win32: homedir tempdir compat

Signed-off-by: Chawye Hsu <su+git@chawyehsu.com>
---
 src/files.c |  4 ++--
 src/utils.c | 12 ++++--------
 2 files changed, 6 insertions(+), 10 deletions(-)

--- a/src/files.c
+++ b/src/files.c
@@ -1462,7 +1462,7 @@ char *check_writable_directory(const char *path)
  * file stream opened in read-write mode.  On error, return NULL. */
 char *safe_tempfile(FILE **stream)
 {
-	const char *env_dir = getenv("TMPDIR");
+	const char *env_dir = getenv("TEMP");
 	char *tempdir = NULL, *tempfile_name = NULL;
 	char *extension;
 	int descriptor;
@@ -1476,7 +1476,7 @@ char *safe_tempfile(FILE **stream)
 		tempdir = check_writable_directory(P_tmpdir);
 
 	if (tempdir == NULL)
-		tempdir = copy_of("/tmp/");
+		tempdir = copy_of("~/AppData/Local/Temp/");
 
 	extension = strrchr(openfile->filename, '.');
 	normalize_slashes(extension);
-- 
2.54.0.windows.1

