diff -ruN nano/src/files.c nanob/src/files.c
--- a/src/files.c	2026-05-15 06:16:54.265033390 -0400
+++ b/src/files.c	2026-05-15 06:19:30.885032362 -0400
@@ -1417,7 +1417,7 @@
  * file stream opened in read-write mode.  On error, return NULL. */
 char *safe_tempfile(FILE **stream)
 {
-	const char *env_dir = getenv("TMPDIR");
+	const char *env_dir = getenv("TEMP");
 	char *tempdir = NULL, *tempfile_name = NULL;
 	char *extension;
 	int descriptor;
@@ -1431,7 +1431,7 @@
 		tempdir = check_writable_directory(P_tmpdir);
 
 	if (tempdir == NULL)
-		tempdir = copy_of("/tmp/");
+		tempdir = copy_of("~/AppData/Local/Temp/");
 
 	extension = strrchr(openfile->filename, '.');
 	normalize_slashes(extension);
diff -ruN nano/src/utils.c nanob/src/utils.c
--- a/src/utils.c	2026-05-15 06:16:54.265033390 -0400
+++ b/src/utils.c	2026-05-15 06:21:35.845031541 -0400
@@ -35,15 +35,11 @@
 	if (homedir == NULL) {
 		const char *homenv = getenv("HOME");
 
-#ifdef HAVE_PWD_H
-		/* When $HOME is unset, or when we're root, try the database. */
-		if (homenv == NULL || geteuid() == ROOT_UID) {
-			const struct passwd *userage = getpwuid(geteuid());
+		if (homenv == NULL || *homenv == '\0')
+			homenv = getenv("USERPROFILE");
 
-			if (userage)
-				homenv = userage->pw_dir;
-		}
-#endif
+		if (homenv == NULL || *homenv == '\0')
+			homenv = getenv("ALLUSERSPROFILE");
 
 		/* Only set `homedir` if a home directory could be determined. */
 		if (homenv && *homenv)
