diff -ruN a/src/files.c b/src/files.c
--- a/src/files.c	2026-06-07 01:06:48.253684926 -0400
+++ b/src/files.c	2026-06-07 01:07:15.353684748 -0400
@@ -174,7 +174,7 @@
 		return FALSE;
 
 	/* Create the lock file -- do not accept an existing one. */
-	fd = open(lockfilename, O_WRONLY|O_CREAT|O_EXCL, RW_FOR_ALL);
+	fd = open(lockfilename, O_WRONLY | _O_BINARY | _O_BINARY|O_CREAT|O_EXCL, RW_FOR_ALL);
 
 	if (fd > 0)
 		filestream = fdopen(fd, "wb");
@@ -256,7 +256,7 @@
 		int lockfd, lockpid, choice;
 		ssize_t readamt;
 
-		if ((lockfd = open(lockfilename, O_RDONLY)) < 0) {
+		if ((lockfd = open(lockfilename, O_RDONLY | _O_BINARY | _O_BINARY)) < 0) {
 			statusline(ALERT, _("Error opening lock file %s: %s"), lockfilename, strerror(errno));
 			free(lockfilename);
 			return NULL;
@@ -863,7 +863,7 @@
 #endif
 
 	/* Try opening the file. */
-	fd = open(full_filename, O_RDONLY);
+	fd = open(full_filename, O_RDONLY | _O_BINARY | _O_BINARY);
 
 #ifndef NANO_TINY
 	restore_handler_for_Ctrl_C();
@@ -1604,7 +1604,7 @@
 	if (unlink(backupname) < 0 && errno != ENOENT && !ISSET(INSECURE_BACKUP))
 		goto problem;
 
-	creation_flags = O_WRONLY|O_CREAT|(ISSET(INSECURE_BACKUP) ? O_TRUNC : O_EXCL);
+	creation_flags = O_WRONLY | _O_BINARY | _O_BINARY|O_CREAT|(ISSET(INSECURE_BACKUP) ? O_TRUNC : O_EXCL);
 
 	/* Create the backup file (or truncate the existing one). */
 	descriptor = open(backupname, creation_flags, S_IRUSR|S_IWUSR);
@@ -1808,7 +1808,7 @@
 #endif
 
 		/* Now open the file.  Use O_EXCL for an emergency file. */
-		descriptor = open(realname, O_WRONLY | O_CREAT | ((method == APPEND) ?
+		descriptor = open(realname, O_WRONLY | _O_BINARY | _O_BINARY | O_CREAT | ((method == APPEND) ?
 					O_APPEND : (normal ? O_TRUNC : O_EXCL)), permissions);
 
 #ifndef NANO_TINY
diff -ruN a/src/nano.c b/src/nano.c
--- a/src/nano.c	2026-06-07 01:06:48.253684926 -0400
+++ b/src/nano.c	2026-06-07 01:07:02.863684830 -0400
@@ -1830,6 +1830,11 @@
 		{NULL, 0, NULL, 0}
 	};
 
+	/* Default file i/o operation to binary mode as linux does.
+	 *  https://learn.microsoft.com/en-us/cpp/c-runtime-library/text-and-binary-mode-file-i-o
+	 */
+	_set_fmode(_O_BINARY);
+
 #ifdef __linux__
 	struct vt_stat dummy;
 
diff -ruN a/src/text.c b/src/text.c
--- a/src/text.c	2026-06-07 01:06:48.253684926 -0400
+++ b/src/text.c	2026-06-07 01:07:16.653684739 -0400
@@ -2361,7 +2361,7 @@
 	/* Fork a process to run spell in. */
 	if ((pid_spell = fork()) == 0) {
 		/* Child: open the temporary file that holds the text to be checked. */
-		if ((tempfile_fd = open(tempfile_name, O_RDONLY)) < 0)
+		if ((tempfile_fd = open(tempfile_name, O_RDONLY | _O_BINARY | _O_BINARY)) < 0)
 			exit(6);
 
 		/* Connect standard input to the temporary file. */
