--- a/src/nano.c.	2026-06-01 04:00:48.937020696 -0400
+++ b/src/nano.c	2026-06-01 04:04:06.907019395 -0400
@@ -835,7 +835,7 @@
 /* Reconnect standard input to the tty, and store its state. */
 void reconnect_and_store_state(void)
 {
-	int thetty = open("/dev/tty", O_RDONLY);
+	int thetty = open("CONIN$", O_RDWR);
 
 	if (thetty < 0 || dup2(thetty, STDIN_FILENO) < 0)
 		die(_("Could not reconnect stdin to keyboard\n"));
@@ -851,17 +851,21 @@
 bool scoop_stdin(void)
 {
 	FILE *stream;
+       int fd;
 
 	restore_terminal();
 
 	/* When input comes from a terminal, show a helpful message. */
 	if (isatty(STDIN_FILENO))
-		fprintf(stderr, _("Reading data from keyboard; type ^D or ^D^D to finish.\n"));
+		fprintf(stderr, _("Reading data from keyboard; type a ^Z line to finish.\n"));
 
 	/* Open standard input. */
-	stream = fopen("/dev/stdin", "rb");
+       fd = dup(0);
+	stream = fdopen(fd, "rb");
 	if (stream == NULL) {
 		int errnumber = errno;
+               if(fd != -1)
+                       close(fd);
 
 		terminal_init();
 		doupdate();
