From 0570efb587e8eb4bbea7abacc21c85a62a16cf20 Mon Sep 17 00:00:00 2001
From: Chawye Hsu <su+git@chawyehsu.com>
Date: Thu, 16 Apr 2026 00:44:35 +0800
Subject: [PATCH 16/17] win32: pdcursesmod ctrlv passthrough

Signed-off-by: Chawye Hsu <su+git@chawyehsu.com>
---
 src/nano.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/nano.c b/src/nano.c
index b751566..a5bba98 100644
--- a/src/nano.c
+++ b/src/nano.c
@@ -1851,6 +1851,13 @@ int main(int argc, char **argv)
 	 */
 	_set_fmode(_O_BINARY);
 
+#ifdef __PDCURSESMOD__
+	/* When using PDCursesMod, we want to disable the default Ctrl+V behavior of
+	 * pasting text from the system clipboard, so that Ctrl+V behaves as a normal
+	 * keystroke that paste text from nano's cutbuffer instead. */
+	PDC_set_function_key(FUNCTION_KEY_PASTE, 0);
+#endif
+
 #ifdef __linux__
 	struct vt_stat dummy;
 
-- 
2.54.0.windows.1

