--- a/doc/nanorc.5
+++ b/doc/nanorc.5
@@ -693,6 +693,9 @@ the line, column, and character positions.
 Counts and reports on the status bar the number of lines, words,
 and characters in the current buffer (or in the marked region).
 .TP
+.B settabsize
+Prompts for a new tabsize to be set.
+.TP
 .B execute
 Prompts for a program to execute.  The program's output will be inserted
 into the current buffer (or into a new buffer when \fBM\-F\fR is toggled).
--- a/src/global.c    2026-04-30 16:47:53.624581654 -0400
+++ b/src/global.c    2026-04-30 16:49:16.051309654 -0400
@@ -674,6 +674,7 @@
                    N_("Count the number of lines, words, and characters");
          const char *suspend_gist = N_("Suspend the editor (return to the shell)");
          const char *setsyntax_gist = N_("Set new syntax highlighting");
+          const char *settabsize_gist = N_("Set new tabsize");
 #endif
          const char *refresh_gist = N_("Refresh (redraw) the current screen");
 #ifdef ENABLE_WORDCOMPLETION
@@ -1031,6 +1032,9 @@

          add_to_funcs(do_set_syntax, MMAIN,
                    N_("Set Syntax"), WHENHELP(setsyntax_gist), TOGETHER);

+          add_to_funcs(do_set_tabsize, MMAIN,
+                    N_("Set Tabsize"), WHENHELP(settabsize_gist), TOGETHER);
 #ifndef NANO_TINY
          add_to_funcs(chop_previous_word, MMAIN,
                              /* TRANSLATORS: The next two strings refer to deleting words. */
@@ -1458,6 +1466,7 @@
          add_to_sclist(MEXECUTE, "^Z", 0, do_suspend, 0);
          add_to_sclist(MMAIN, "^Z", 0, suggest_ctrlT_ctrlZ, 0);
          add_to_sclist(MMAIN, "M-D", 0, count_lines_words_and_characters, 0);
+          add_to_sclist(MMAIN, "M-4", 0, do_set_tabsize, 0);
          add_to_sclist(MMAIN, "M-5", 0, do_set_syntax, 0);
 #else
          add_to_sclist(MMAIN, "M-H", 0, do_help, 0);
--- a/src/prototypes.h
+++ b/src/prototypes.h
@@ -519,6 +519,7 @@ void do_formatter(void);
 #endif
 #ifndef NANO_TINY
 void count_lines_words_and_characters(void);
+void do_set_tabsize(void);
 #endif
 void do_verbatim_input(void);
 void complete_a_word(void);
--- a/src/rcfile.c
+++ b/src/rcfile.c
@@ -321,6 +321,8 @@ keystruct *strtosc(const char *input)
 		s->func = do_find_bracket;
 	else if (!strcmp(input, "wordcount"))
 		s->func = count_lines_words_and_characters;
+	else if (!strcmp(input, "settabsize"))
+		s->func = do_set_tabsize;
 	else if (!strcmp(input, "recordmacro"))
 		s->func = record_macro;
 	else if (!strcmp(input, "runmacro"))
--- a/src/text.c
+++ b/src/text.c
@@ -2969,6 +2969,28 @@ void count_lines_words_and_characters(void)
 						words, P_("word", "words", words),
 						chars, P_("character", "characters", chars));
 }
+
+/* Prompt user to set the new tabsize. We use the spell menu because
+ * it has no functions. */
+void do_set_tabsize(void)
+{
+	ssize_t new_tabsize = -1;
+	int response = do_prompt(MSPELL, "", NULL, edit_refresh, "New tabsize");
+
+	/* Cancel if no answer provided. */
+	if (response != 0) {
+		statusbar(_("Cancelled"));
+		return;
+	}
+
+	if (!parse_num(answer, &new_tabsize) || new_tabsize <= 0) {
+		statusline(AHEM, _("Requested tab size \"%s\" is invalid"), answer);
+		return;
+	}
+
+	tabsize = new_tabsize;
+	statusline(REMARK, _("Tabsize set to %d"), tabsize);
+}
 #endif /* !NANO_TINY */
 
 /* Get verbatim input. */
-- 
2.30.2
