--- a/curses/wingui/Makefile	2026-05-16 20:38:12.594126630 -0400
+++ b/curses/wingui/Makefile	2026-05-17 01:07:22.574020573 -0400
@@ -23,7 +23,7 @@
 
 CAT		= cat
 PREFIX		=
-PATH_SEP		= /
+PATH_SEP	= /
 CP		= cp
 RM		= rm -f
 
@@ -56,13 +56,18 @@
 # Only decision is:  are we doing a 64-bit compile (_w64 defined)?
 
 ifndef ON_WINDOWS
-	PREFIX  = i686-w64-mingw32-
+	ifdef _w32
+            	PREFIX  = i686-w64-mingw32-
+	endif
 	ifdef _w64
 		PREFIX  = x86_64-w64-mingw32-
 	endif
 	ifdef _a64
 		PREFIX  = aarch64-w64-mingw32-
 	endif
+	ifdef _a32
+		PREFIX  = armv7-w64-mingw32-
+	endif
 endif
 
 PDCURSES_WIN_H	= $(osdir)/pdcwin.h
@@ -77,16 +82,24 @@
 	CFLAGS  += -g -DPDCDEBUG
 	LDFLAGS = -g
 else
-	CFLAGS  += -O3
+	CFLAGS  += -O3 -DPDC_GUI
 	LDFLAGS =
 endif
 
 CFLAGS += -I$(PDCURSES_SRCDIR)
 
+ifdef CHTYPE_64
+	CFLAGS += -DCHTYPE_64
+endif
+
 ifdef CHTYPE_32
 	CFLAGS += -DCHTYPE_32
 endif
 
+ifdef CHTYPE_16
+	CFLAGS += -DCHTYPE_16
+endif
+
 ifeq ($(UTF8),Y)
 	CFLAGS	+= -DPDC_FORCE_UTF8
 else
