--- a/curses/pdcurses/addch.c   2026-05-10 00:00:00.000000000 -0400
+++ b/curses/pdcurses/addch.c   2026-05-10 00:00:00.000000000 -0400
@@ -537,8 +537,11 @@
     {
         const bool is_combining = (text && !text_width);
+        const bool is_zwj = (text == 0x200D);
+        const bool is_variation_selector = (text >= 0xFE00 && text <= 0xFE0F);
+        const bool is_skin_tone_modifier = (text >= 0x1F3FB && text <= 0x1F3FF);
         const bool is_low_surrogate = PDC_IS_LOW_SURROGATE( text);

-        if( is_combining || is_low_surrogate)
+        if( is_combining || is_zwj || is_variation_selector || is_skin_tone_modifier || is_low_surrogate)
         {
             chtype prev_char;

@@ -553,7 +556,7 @@
                 x = win->_maxx - 1;
             }

-            if( is_combining)
+            if( is_combining || is_zwj || is_variation_selector || is_skin_tone_modifier)
                 text = COMBINED_CHAR_START
                          + PDC_find_combined_char_idx( prev_char, text);
             else if( PDC_IS_HIGH_SURROGATE( prev_char))
