--- a/curses/wingui/pdcdisp.c	2026-05-17 21:12:24.385456340 +0000
+++ b/curses/wingui/pdcdisp.c	2026-05-17 21:12:24.408168186 +0000
@@ -368,6 +368,12 @@
 #endif
         return;
     }
+    if( lineno < 0 || lineno >= SP->lines || x < 0 || x >= SP->cols || len <= 0)
+        return;
+    if( len > SP->cols - x)
+        len = SP->cols - x;
+    if( len <= 0)
+        return;
     assert( x >= 0);
     assert( len <= SP->cols - x);
     assert( lineno < SP->lines && len > 0 && lineno >= 0);
