--- a/curses/pdcurses/getch.c
+++ b/curses/pdcurses/getch.c
@@ -550,7 +550,7 @@
     /* refresh window when wgetch is called if there have been changes
        to it and it is not a pad */
 
-    if (!(win->_flags & _PAD) && ((!win->_leaveit &&
+    if (!SP->resized && !(win->_flags & _PAD) && ((!win->_leaveit &&
          (win->_begx + win->_curx != SP->curscol ||
           win->_begy + win->_cury != SP->cursrow)) || is_wintouched(win)))
         wrefresh(win);
@@ -568,7 +569,5 @@
     if( key != ERR)
     {
-        if( key == KEY_RESIZE)
-            resize_term( 0, 0);
         return( key);
     }
     /* prepare to buffer data */
@@ -663,7 +662,5 @@
         if( SP->cbreak)
         {
-            if( key == KEY_RESIZE)
-                resize_term( 0, 0);
             return key;
         }
 
@@ -683,8 +680,6 @@
         if (key == '\n' || key == '\r')
         {
-            if( SP->c_buffer[SP->c_gindex] == KEY_RESIZE)
-                resize_term( 0, 0);
             return SP->c_buffer[SP->c_gindex++];
         }
     }
