--- a/curses/wingui/pdcdisp.c
+++ b/curses/wingui/pdcdisp.c
@@ -589,8 +589,14 @@
         const HDC hdc = (override_hdc ? override_hdc : GetDC( PDC_hWnd)) ;
 
         assert( len);
 #ifdef PDC_WIDE
+        while( len && (srcp[len - 1] & A_CHARTEXT) == MAX_UNICODE)
+            len--;
+        if( !len)
+        {
+            ReleaseDC( PDC_hWnd, hdc);
+            return;
+        }
         assert( (srcp[len - 1] & A_CHARTEXT) != MAX_UNICODE);
 #endif
         PDC_transform_line_given_hdc( hdc, lineno, x, len, srcp);
