--- a/curses/wincon/Makefile	2026-05-16 04:38:18.654504849 -0400
+++ b/curses/wincon/Makefile	2026-05-16 04:58:46.284496787 -0400
@@ -23,7 +22,7 @@
 
 CAT		= cat
 PREFIX		=
-PATH_SEP		= /
+PATH_SEP	= /
 CP		= cp
 RM		= rm -f
 
@@ -56,12 +55,21 @@
 # Only decision is:  are we doing a 64-bit compile (_w64 defined)?
 
 ifndef ON_WINDOWS
-	PREFIX  = i686-w64-mingw32-
+	ifdef _w32
+		PREFIX  = i686-w64-mingw32-
+		NOT_WIN = Y
+ 	endif
 	ifdef _w64
		PREFIX  = x86_64-w64-mingw32-
+		NOT_WIN = Y
 	endif
 	ifdef _a64
		PREFIX  = aarch64-w64-mingw32-
+		NOT_WIN = Y
+	endif
+	ifdef _a32
+		PREFIX = armv7-w64-mingw32-
+		NOT_WIN = Y
 	endif
 endif
 
@@ -77,16 +84,35 @@
 	CFLAGS  += -g -DPDCDEBUG
 	LDFLAGS = -g
 else
-	CFLAGS  += -O3
+	CFLAGS  += -DPDC_WINCON
 	LDFLAGS =
 endif
 
+ifeq ($(DOS),Y)
+	CFLAGS += -DDOS
+	RM = del
+endif
+
+ifeq ($(NOT_WIN),Y)
+	CFLAGS += -fPIC
+	EXELIBS = -lwinmm
+	RM = rm
+endif
+
 CFLAGS += -I$(PDCURSES_SRCDIR)
 
+ifdef CHTYPE_64
+	CFLAGS += -DCHTYPE_64
+endif
+
 ifdef CHTYPE_32
 	CFLAGS += -DCHTYPE_32
 endif
 
+ifdef CHTYPE_16
+	CFLAGS += -DCHTYPE_16
+endif
+
 ifeq ($(UTF8),Y)
 	CFLAGS += -DPDC_FORCE_UTF8
 else
@@ -99,7 +126,7 @@
 	PDCCFLAGS += -DHAVE_NO_INFOEX
 endif
 
-LINK	   = $(CC)
+LINK	 = $(CC)
 LIBNAME = pdcurses
 DLLNAME = pdcurses
 
