--- a/curses/vt/Makefile	2026-05-16 00:56:20.234592311 -0400
+++ b/curses/vt/Makefile	2026-05-16 00:56:12.314592363 -0400
@@ -19,7 +19,7 @@
 
 RM		= rm -f
 
-CFLAGS  = -Wall -Wextra -pedantic -Werror
+CFLAGS  = -Wall -Wextra -pedantic
 ifeq ($(DEBUG),Y)
 	CFLAGS  += -g -DPDCDEBUG
 else
@@ -40,6 +40,10 @@
 	endif
 endif
 
+ifdef CHTYPE_64
+       CFLAGS += -DCHTYPE_64
+endif
+
 ifdef CHTYPE_32
 	CFLAGS += -DCHTYPE_32
 endif
@@ -72,7 +76,7 @@
 	E = .exe
 endif
 
-LIBNAME=libpdcurses
+LIBNAME=pdcurses
 DLLNAME=pdcurses
 
 LIBFLAGS = rv
