--- Makefile	2026-05-03 12:40:26.649456789 -0400
+++ Makefile.p	2026-05-03 16:21:18.820348493 -0400
@@ -19,7 +19,7 @@
 
 RM		= rm -f
 
-CFLAGS  = -Wall -Wextra -pedantic -Werror
+CFLAGS  = -Wall -Wextra -Wno-error
 ifeq ($(DEBUG),Y)
 	CFLAGS  += -g -DPDCDEBUG
 else
@@ -51,16 +51,19 @@
 ifdef _w64
 	PREFIX  = x86_64-w64-mingw32-
 	IS_WIN = Y
+          CFLAGS += -DPDC_VT
 endif
 
 ifdef _w32
 	PREFIX  = i686-w64-mingw32-
 	IS_WIN = Y
+          CFLAGS += -DPDC_VT
 endif
 
 ifdef _a64
 	PREFIX  = aarch64-w64-mingw32-
 	IS_WIN = Y
+          CFLAGS += -DPDC_VT
 endif
 
 ifeq ($(OS),Windows_NT)
@@ -72,7 +75,7 @@
 	E = .exe
 endif
 
-LIBNAME=libpdcurses
+LIBNAME=pdcurses
 DLLNAME=pdcurses
 
 LIBFLAGS = rv
@@ -94,7 +97,7 @@
 else ifeq ($(TARGET_OS),msdosdjgpp)
 	CFLAGS += -DDOS
 	E = .exe
-else
++else ifneq ($(IS_WIN),Y)
 	CFLAGS += -fPIC
 endif
 
