--- a/curses/wincon/pdcdisp.c
+++ b/curses/wincon/pdcdisp.c
@@ -155,7 +155,7 @@
 #endif
 
 #ifdef PDC_WIDE
-#define IS_SUPPLEMENTAL_MULTILINGUAL_PLANE( c) ((c) & 0x1f0000)
+#define IS_SUPPLEMENTAL_MULTILINGUAL_PLANE(c) ((c) >= 0x10000 && (c) <= 0x10FFFF)
 #endif
 
 static void _show_run_of_ansi_characters( const attr_t attr,
@@ -181,8 +181,6 @@
 
 #ifdef PDC_WIDE
         ch &= A_CHARTEXT;
-        if( ch <= MAX_UNICODE)
-        {
             if( IS_SUPPLEMENTAL_MULTILINGUAL_PLANE( ch))
             {
                 buffer[n_out++] = (WCHAR)((ch - 0x10000) >> 10 | 0xD800); /* first UTF-16 unit */
@@ -190,7 +188,6 @@
             }
         else
             buffer[n_out++] = (WCHAR)ch;
-        }
 #else
         buffer[n_out++] = (char)( ch & A_CHARTEXT);
 #endif
@@ -269,10 +266,8 @@
             }
         }
 #endif
-        if( ch <= MAX_UNICODE)
-        {
             if( IS_SUPPLEMENTAL_MULTILINGUAL_PLANE( ch))
             {
                 buffer[n_out++].Char.UnicodeChar = (WCHAR)((ch - 0x10000) >> 10 | 0xD800); /* first UTF-16 unit */
                 buffer[n_out++].Char.UnicodeChar = (WCHAR)(ch & 0x3FF) | 0xDC00;   /* second UTF-16 unit */
             }
             else
                 buffer[n_out++].Char.UnicodeChar = (WCHAR)ch;
-        }
 #else
         buffer[n_out++].Char.UnicodeChar = (WCHAR)( ch & A_CHARTEXT);
 #endif
	