--- a/curses/wincon/pdcdisp.c	2026-05-01 19:56:54.114523945 -0400
+++ b/curses/wincon/pdcdisp	2026-05-01 20:21:35.257713145 -0400
@@ -155,7 +155,7 @@
 #endif
 
 #ifdef PDC_WIDE
-#define IS_SUPPLEMENTAL_MULTILINGUAL_PLANE( c) ((c) & 0x1f0000)
+#define IS_SUPPLEMENTAL_MULTILINGUAL_PLANE(c) ((c) >= 0x10000 && (c) <= 0x10FFFF)
 #endif
 
 static void _show_run_of_ansi_characters( const attr_t attr,
@@ -163,7 +163,7 @@
                            const int lineno, const int x, const chtype *srcp, const int len)
 {
 #ifdef PDC_WIDE
-    WCHAR buffer[MAX_PACKET_LEN];
+    WCHAR buffer[MAX_PACKET_LEN * 2];
 #else
     char buffer[MAX_PACKET_LEN];
 #endif
@@ -181,17 +181,17 @@
 
 #ifdef PDC_WIDE
         ch &= A_CHARTEXT;
-        if( ch <= MAX_UNICODE)
+    if (ch > 0xFFFF && ch <= 0x10FFFF) 
         {
-            if( IS_SUPPLEMENTAL_MULTILINGUAL_PLANE( ch))
-            {
-                buffer[n_out++] = (WCHAR)((ch - 0x10000) >> 10 | 0xD800); /* first UTF-16 unit */
-                buffer[n_out++] = (WCHAR)(ch & 0x3FF) | 0xDC00;   /* second UTF-16 unit */
+        buffer[n_out++] = (WCHAR)(((ch - 0x10000) >> 10) | 0xD800);
+        buffer[n_out++] = (WCHAR)(((ch - 0x10000) & 0x3FF) | 0xDC00);
             }
         else
+    {
             buffer[n_out++] = (WCHAR)ch;
         }
 #else
+    /* This ensures non-wide builds (like some legacy m68k targets) still work */
         buffer[n_out++] = (char)( ch & A_CHARTEXT);
 #endif
     }
@@ -260,8 +260,6 @@
             }
         }
 #endif
-        if( ch <= MAX_UNICODE)
-        {
             if( IS_SUPPLEMENTAL_MULTILINGUAL_PLANE( ch))
             {
                 buffer[n_out++].Char.UnicodeChar = (WCHAR)((ch - 0x10000) >> 10 | 0xD800); /* first UTF-16 unit */
@@ -269,7 +267,6 @@
             }
             else
                 buffer[n_out++].Char.UnicodeChar = (WCHAR)ch;
-        }
 #else
         buffer[n_out++].Char.UnicodeChar = (WCHAR)( ch & A_CHARTEXT);
 #endif
