--- a/curses/pdcurses/addch.c   2026-05-10 00:00:00.000000000 -0400
+++ b/curses/pdcurses/addch.c   2026-05-10 00:00:00.000000000 -0400
@@ -537,8 +537,31 @@
     {
         const bool is_combining = (text && !text_width);
         const bool is_low_surrogate = PDC_IS_LOW_SURROGATE( text);
+        bool join_with_previous = FALSE;
+
+        if( !is_combining && !is_low_surrogate && (x || y))
+        {
+            int px = x;
+            int py = y;
+            chtype prev_char;
+
+            if( px)
+                px--;
+            else
+            {
+                py--;
+                px = win->_maxx - 1;
+            }
+            prev_char = win->_y[py][px] & A_CHARTEXT;
+            if( prev_char == 0x200D)
+                join_with_previous = TRUE;
+            else if( prev_char > MAX_UNICODE)
+            {
+                cchar_t last_added;
+                (void)PDC_expand_combined_characters( prev_char, &last_added);
+                if( last_added == 0x200D)
+                    join_with_previous = TRUE;
+            }
+        }
 
-        if( is_combining || is_low_surrogate)
+        if( is_combining || is_low_surrogate || join_with_previous)
         {
             chtype prev_char;
 
@@ -553,6 +576,9 @@
             }
             prev_char = win->_y[y][x] & A_CHARTEXT;
+            if( join_with_previous)
+                text_width = 0;
             if( is_combining)
                 text = COMBINED_CHAR_START
                          + PDC_find_combined_char_idx( prev_char, text);
