--- a/curses/vt/Makefile	2026-05-02 08:14:28.763836238 -0400
+++ b/curses/vt/Makefile	2026-05-02 09:22:55.640534267 -0400
@@ -51,16 +51,19 @@
 ifdef _w64
 	PREFIX  = x86_64-w64-mingw32-
 	IS_WIN = Y
+	CFLAGS += -DPDC_VT
 endif
 
 ifdef _w32
 	PREFIX  = i686-w64-mingw32-
 	IS_WIN = Y
+	CFLAGS += -DPDC_VT
 endif
 
 ifdef _a64
 	PREFIX  = aarch64-w64-mingw32-
 	IS_WIN = Y
+	CFLAGS += -DPDC_VT
 endif
 
 ifeq ($(OS),Windows_NT)
@@ -72,7 +75,7 @@
 	E = .exe
 endif
 
-LIBNAME=libpdcurses
+LIBNAME=pdcurses
 DLLNAME=pdcurses
 
 LIBFLAGS = rv
@@ -94,7 +97,7 @@
 else ifeq ($(TARGET_OS),msdosdjgpp)
 	CFLAGS += -DDOS
 	E = .exe
-else
+else ifneq ($(IS_WIN),Y)
 	CFLAGS += -fPIC
 endif
 
